### PR TITLE
:boom: react context

### PR DESCRIPTION
**Pull request by fabienjuif**
closes #139 
closes #140 
closes #141 
closes #142  

----

 - This PR uses new ReactJS Context API and new ReactJS lifecycle.
 - `inject` is called only when state is not the same as before (or props changed)
 - There is no heap difference (I think TodoMVC is a tiny application) (Snapshot 1 == new // Snapshot 2 == old)
image

----

I just understand that `getDerivedStateFromProps` is static.
So I need to move context to the new API to avoid using `this` before completing this PR.
This PR doesn't event "compile" this is a raw WIP.

### CODE REVIEW

File: packages/connectors/react/src/inject.jsx [hunk 1]
```diff
@@ -21,16 +21,18 @@ export default injectFunction => WrappedComponent => class extends Component {
...
     super(props, context)
-
     this.first = true
-    this.state = {
-      injectedProps: {},
-    }
-  }
 
-  componentWillMount() {
+    // attach store
```
Comment (fabienjuif) [inline packages/connectors/react/src/inject.jsx]: move it

**Comment by EmrysMyrddin** [comment]
This means that it k-ramel will no longer work on previous react version ?

**Comment by fabienjuif** [comment]
I don't now yet :)

**Comment by fabienjuif** [comment]
@EmrysMyrddin I think I will do the update.

> If you don't use the latest XXX lib, then you can use an older version of the lib.
> If you want the latest bug fixes and features, you should upgrade dependencies.

What do you think about this ?

**Comment by fabienjuif** [comment]
- [ ] Provider should insert into context (like before) for older libs (older hoc)
- [ ] Tests
- [ ] Something to have a nicer React tree ?

image

**Comment by fabienjuif** [comment]
I close this PR, we will write a new one when we will be ready and motivated

**Pull request closed without merging**